lib/rubocop/cop/autocorrect_logic.rb

```diff
@@ -5,7 +5,9 @@ module Cop
     # This module encapsulates the logic for autocorrect behaviour for a cop
     module AutocorrectLogic
       def autocorrect?
-        @options[:auto_correct] && support_autocorrect? && autocorrect_enabled?
+        return nil unless support_autocorrect?
```

Comment:
Boolean methods should return either `true` or `false`. Returning anything else is unexpected/uncommon.